coreos-koji-tagger/coreos_koji_tagger.py

```diff
@@ -314,6 +326,23 @@ def process_github_push_message(self, message: fedora_messaging.api.Message):
 
         self.process_git_lockfiles(commit)
 
+    @catch_exceptions_and_continue
+    def process_coreos_stream_release_message(self, message: fedora_messaging.api.Message):
+        msg = message.body
+        stream = msg['stream']
+
+        # we only handle rawhide and branched builds; see
+        # https://github.com/coreos/fedora-coreos-releng-automation/issues/181
+        if stream not in ['rawhide', 'branched']:
```

Comment:
Operating on `rawhide` now might mean we need to change our SOP for updating the signatures for our pool. See https://github.com/coreos/fedora-coreos-tracker/blob/fe29ffc877c504482dc9bd6e17a1ed9a79eeb628/.github/ISSUE_TEMPLATE/rebase.md?plain=1#L30-L32

Reply:
We also may want to be more aggressive about pruning rawhide rpms since this will cause the size of our pool to rise dramatically (IMO), which will have a side effect of causing the distrepo tasks to take longer.

Reply:
> Operating on `rawhide` now might mean we need to change our SOP for updating the signatures for our pool. See [coreos/fedora-coreos-tracker@`fe29ffc`/.github/ISSUE_TEMPLATE/rebase.md?plain=1#L30-L32](https://github.com/coreos/fedora-coreos-tracker/blob/fe29ffc877c504482dc9bd6e17a1ed9a79eeb628/.github/ISSUE_TEMPLATE/rebase.md?plain=1#L30-L32)

Isn't that command already doing what we need? (It's adding the rawhide key after rawhide moves to N+1). E.g. right now, the coreos-pool tag has the rawhide key in its `tag2distrepo.keys`.

> We also may want to be more aggressive about pruning rawhide rpms since this will cause the size of our pool to rise dramatically (IMO), which will have a side effect of causing the distrepo tasks to take longer.

Yes, definitely. I think one thing we can do here is at branching time, we reset everything: for Bodhi testing of rawhide, we don't want to use a e.g. f40 lockfile to test an f41 package since it might not even depsolve, and for Bodhi testing of branched, I think it'll be too complex to have the code know to take the last successful rawhide build and test from that, so it'd make more sense to just not use the "auto-lockfile" behaviour there until we get a clean branched build to base of off going forward.

Reply:
> > Operating on `rawhide` now might mean we need to change our SOP for updating the signatures for our pool. See [coreos/fedora-coreos-tracker@`fe29ffc`/.github/ISSUE_TEMPLATE/rebase.md?plain=1#L30-L32](https://github.com/coreos/fedora-coreos-tracker/blob/fe29ffc877c504482dc9bd6e17a1ed9a79eeb628/.github/ISSUE_TEMPLATE/rebase.md?plain=1#L30-L32)
> 
> Isn't that command already doing what we need? (It's adding the rawhide key after rawhide moves to N+1). E.g. right now, the coreos-pool tag has the rawhide key in its `tag2distrepo.keys`.

Ahh yes.. I forgot we moved it earlier and it covers rawhide now..  Ignore me.

> 
> > We also may want to be more aggressive about pruning rawhide rpms since this will cause the size of our pool to rise dramatically (IMO), which will have a side effect of causing the distrepo tasks to take longer.
> 
> Yes, definitely. I think one thing we can do here is at branching time, we reset everything: for Bodhi testing of rawhide, we don't want to use a e.g. f40 lockfile to test an f41 package since it might not even depsolve, and for Bodhi testing of branched, I think it'll be too complex to have the code know to take the last successful rawhide build and test from that, so it'd make more sense to just not use the "auto-lockfile" behaviour there until we get a clean branched build to base of off going forward.

I think ther are some details here I'm not quite fully groking but we can dig into that later.

Reply:
Sorry, "we reset everything" is quite vague. To be more specific, I mean "we untag all rawhide packages".